media-sound/supercollider: bump to 3.9.3

simonvanderveldt opened the pull request:
Did some minor cleanup of the existing 3.8.0 ebuild since `app-misc/cwiid` has been removed, `-DNATIVE=ON` should've never been in the ebuild and the `required_use` wasn't necessary.
There are more things I'd like to fix, like being able to build with system boost and yaml-cpp, but both currently cause a build failure, see https://github.com/supercollider/supercollider/issues/3513#issuecomment-398902784

I wasn't sure how to handle the patches, since they need to be used by multiple versions of the ebuild but were versioned for 3.8.0.
For now I just used the existing patch filenames, though it might make more sense to remove the version number from the patch filenames. Just let me know what's preferred and I'll update this PR.

Even though I can't reproduce it, this probably fixes https://bugs.gentoo.org/615306
Validated working by building all of these in a clean amd64 stage3.

a17r commented:
No one is going to use the versions between 3.8.0 and 3.9.3. If there is already a .3 point release available, there is really no point to make the versions with unsolved bugs available.

> I wasn't sure how to handle the patches, since they need to be used by multiple versions of the ebuild but were versioned for 3.8.0.

You keep the version number of the first version the patch was introduced for.

simonvanderveldt commented:
@a17r Thanks for the feedback! I've removed the initial 3.9 releases and kept the 3.8.1 release because it fixes 2 bugs in 3.8.0 and the ABI has changed for 3.9 which means there might be some plugins that aren't compatible with 3.9 yet.

Also removed the `wiimote` useflag from `metadata.xml`

a17r commented on media-sound/supercollider/supercollider-3.8.0-r1.ebuild:
```diff
@@ -65,7 +63,6 @@ src_configure() {
 	local mycmakeargs=(
 		-DAUDIOAPI=$(usex jack jack portaudio)
 		-DINSTALL_HELP=ON
-		-DNATIVE=ON
```
You are changing the install image without making a revbump. I suspect you make the same change in 3.8.1 and 3.9.4, so if .0 was wrong in that regard, simply remove the old ebuild in a separate commit.

simonvanderveldt replied:
@a17r Do you think that change is worth a revbump/recompile? I wasn't sure, it was wrong of course, but not sure if it would be worth a recompile.
I can drop the change to the 3.8.0-r1 ebuild as well. What do you prefer?

a17r replied:
well, the point was that you drop the changes to -r1, and drop the ebuild altogether after those bumps.

simonvanderveldt replied:
@a17r Just pushed the changes, hope it's OK like this :)

a17r commented:
thanks!
